associations/views/association.py

```diff
@@ -56,7 +64,21 @@ class AssociationViewSet(viewsets.ModelViewSet):
     permission_classes = (AssociationPermission,)
 
     def get_serializer_class(self):
-        if self.action in ('list',):
+        if self.action in ("list",):
             return AssociationShortSerializer
         else:
             return AssociationSerializer
+
+
+@api_view(["POST"])
+def set_association_image(request, association_pk):
+    association = Association.objects.get(pk=association_pk)
+    role = request.user.get_role(association)
+    if not (role and role.administration):
+        raise PermissionDenied()
+
+    serializer = AssociationImageSerializer(data=request)
+    serializer.is_valid(raise_exception=True)
+    serializer.update(association, serializer.validated_data)
+
+    return Response(status=200)
```

Comment:
Every time I test a create function, the default response of DRF is a 201, so I believe it should be the same here

Reply:
this is not a creation, but an update. Maybe the code should not be 200, but I beleive it's not 201 either

Reply:
Actually, maybe the http verb to use is `PUT`.
Also see the MDN for the response codes (yeah, there are several cases…) https://developer.mozilla.org/fr/docs/Web/HTTP/M%C3%A9thode/PUT#R%C3%A9ponses